controllers/flp/flp_pipeline_builder.go

```diff
@@ -480,7 +483,57 @@ func getIPFIXTransport(transport string) string {
 	}
 }
 
-func getKafkaTLS(tls *flowslatest.ClientTLS, volumeName string, volumes *volumes.Builder) *api.ClientTLS {
+func (b *PipelineBuilder) createOpenTelemetryStage(name string, spec *flowslatest.FlowCollectorOpenTelemetry, fromStage *config.PipelineBuilderStage, flpMetrics []api.MetricsItem) {
+	conn := api.OtlpConnectionInfo{
+		Address:        spec.TargetHost,
+		Port:           spec.TargetPort,
+		ConnectionType: getOtelConnType(spec.Type),
+		TLS:            getClientTLS(&spec.TLS, name, b.volumes),
+		Headers:        spec.Headers,
+	}
+
+	// otel logs config
+	if spec.Logs.Enable != nil && *spec.Logs.Enable {
+		transformConfig := api.TransformGeneric{
+			Policy: "replace_keys",
+			Rules:  constants.OpenTelemetryDefaultTransformRules,
+		}
+		// set custom rules if specified
+		if spec.Rules != nil {
+			transformConfig.Rules = *spec.Rules
+		}
+		// add transform stage
+		transformStage := fromStage.TransformGeneric(fmt.Sprintf("%s-transform", name), transformConfig)
+		// add encode stage(s)
+		transformStage.EncodeOtelLogs(fmt.Sprintf("%s-logs", name), api.EncodeOtlpLogs{
+			OtlpConnectionInfo: &conn,
+		})
+	}
+
+	// otel metrics config
+	if spec.Metrics.Enable != nil && *spec.Metrics.Enable {
+		fromStage.EncodeOtelMetrics(fmt.Sprintf("%s-metrics", name), api.EncodeOtlpMetrics{
+			OtlpConnectionInfo: &conn,
+			Prefix:             constants.OperatorName,
+			Metrics:            flpMetrics,
```

Comment:
:point_up: we should probably adapt the metrics as same as logs to match Open Telemetry field namings

For now I just got rid of the transform stage and kept the same metrics as prometheus. Let me know your thoughts

Reply:
+1, that means to rewrite `flpMetrics` to map all the labels and the filters with the otel semantic.
That can be done in a follow-up I guess, if you prefer

Reply:
Sure I can add a todo

Reply:
done in https://github.com/netobserv/network-observability-operator/pull/671/commits/6b556fbf0ab08357fa0a1b6033ba9d61508bd06b